datafusion-postgres/src/pg_catalog/context.rs

```diff
@@ -0,0 +1,121 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait PgCatalogContextProvider: Clone + Debug + Send + Sync + 'static {
```

Comment:
`PgCatalogContextProvider` methods are all related to the auth use case, but the name is much more general.  Is the idea that `PgCatalogContextProvider` could expand to provide other (non auth related) context in future?